### PR TITLE
Adding companieshouse and postcodelookup connectors

### PR DESCRIPTION
**Pull request by irham0019**

### CODE REVIEW

File: companieshouse/companieshouse-1.0.0/org.wso2.carbon.connector/src/main/resources/company/component.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied. See the License for the
+   specific language governing permissions and limitations
+   under the License.
+-->
+<component name="company" type="synapse/template" >
+    <subComponents> 
+		<component name="companieshouselookup" >
+	    		<file>companieshouselookup.xml</file>
+	    		<description>sample wso2 connector method</description>
```
Comment (malakasilva) [inline companieshouse/companieshouse-1.0.0/org.wso2.carbon.connector/src/main/resources/company/component.xml]: Change the description

File: postcodelookup/postcodelookup-1.0.0/org.wso2.carbon.connector/src/main/resources/postCodeLookup/component.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied. See the License for the
+   specific language governing permissions and limitations
+   under the License.
+-->
+<component name="postcodelookup" type="synapse/template" >
+    <subComponents> 
+		<component name="init" >
+	    		<file>postcodelookup.xml</file>
+	    		<description>sample wso2 connector method</description>
```
Comment (malakasilva) [inline postcodelookup/postcodelookup-1.0.0/org.wso2.carbon.connector/src/main/resources/postCodeLookup/component.xml]: Change the description

File: postcodelookup/postcodelookup-1.0.0/org.wso2.carbon.connector/pom.xml [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>org.wso2.carbon.connector</artifactId>
+
+    <groupId>org.wso2.carbon.connector</groupId>
+    <packaging>bundle</packaging>
+    <version>1.0.0</version>
+    <name>WSO2 Carbon - Connector For postCodeLookup</name>
+    <url>http://wso2.org</url>
+<!--
```
Comment (malakasilva) [inline postcodelookup/postcodelookup-1.0.0/org.wso2.carbon.connector/pom.xml]: Remove commented code

File: companieshouse/companieshouse-1.0.0/org.wso2.carbon.connector/src/main/resources/company/companieshouselookup.xml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        <parameter name="startIndex" description="start index"/>
+        <sequence>
+            <property name="uri.var.company" expression="$func:company"/>
+            <property name="uri.var.itemsPerPage" expression="$func:itemsPerPage"/>
+            <property name="uri.var.startIndex" expression="$func:startIndex"/>
+            <script language="js"><![CDATA[
+            var itemsPerPage=mc.getProperty('uri.var.itemsPerPage');
+            var startIndex=mc.getProperty('uri.var.startIndex');
+            var query="&";
+            if(itemsPerPage != null && itemsPerPage != ""){
```
Comment (pamod) [inline companieshouse/companieshouse-1.0.0/org.wso2.carbon.connector/src/main/resources/company/companieshouselookup.xml]: As an improvement can't we just do the validation if(itemsPerPage) ? You could refer http://stackoverflow.com/questions/5515310/is-there-a-standard-function-to-check-for-null-undefined-or-blank-variables-in for more details

File: postcodelookup/postcodelookup-1.0.0/org.wso2.carbon.connector/src/main/resources/config/init.xml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied. See the License for the
+   specific language governing permissions and limitations
+   under the License.
+-->
+<template name="init" onError="fault" xmlns="http://ws.apache.org/ns/synapse">
+   <parameter name="apiUrl" description="api url of the API"/>
+   <parameter name="apiKey" description="api key of the API"/>
```
Comment (pamod) [inline postcodelookup/postcodelookup-1.0.0/org.wso2.carbon.connector/src/main/resources/config/init.xml]: Can't we be more precise with the description ? i.e API key is used for authentication which will be encoded etc

File: postcodelookup/postcodelookup-1.0.0/org.wso2.carbon.connector/src/main/resources/postCodeLookup/postcodelookup.xml [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+            var exclude=mc.getProperty('uri.var.exclude');
+            var format=mc.getProperty('uri.var.format');
+            var addtags=mc.getProperty('uri.var.addtags');
+            var identifier=mc.getProperty('uri.var.identifier');
+            var callback=mc.getProperty('uri.var.callback');
+            var page=mc.getProperty('uri.var.page');
+            var postcodeonly=mc.getProperty('uri.var.postcodeonly');
+            var query="?";
+
+            if(responseType != null && responseType != ""){
```
Comment (pamod) [inline postcodelookup/postcodelookup-1.0.0/org.wso2.carbon.connector/src/main/resources/postCodeLookup/postcodelookup.xml]: As a suggestion, could we improve this further, concise it by adding all the elements to an array at the initial stage and do the string concatenation through a for-each loop ? , The reason is if the API post code lookup service keeps changing by adding/ omitting query params, having a lot of conditional statements would be hard to maintain.

File: companieshouse/companieshouse-1.0.0/org.wso2.carbon.connector/src/main/resources/company/component.xml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied. See the License for the
+   specific language governing permissions and limitations
+   under the License.
+-->
+<component name="company" type="synapse/template" >
+    <subComponents> 
+		<component name="searchcompanies" >
+	    		<file>searchcompanies.xml</file>
```
Comment (ashalya) [inline companieshouse/companieshouse-1.0.0/org.wso2.carbon.connector/src/main/resources/company/component.xml]: searchcompanies should be searchCompanies

File: companieshouse/companieshouse-1.0.0/org.wso2.carbon.connector/src/main/resources/company/searchcompanies.xml [hunk 8]
```diff
@@ -0,0 +1,48 @@
+<!--
```
Comment (ashalya) [inline companieshouse/companieshouse-1.0.0/org.wso2.carbon.connector/src/main/resources/company/searchcompanies.xml]: method name should be camel case.

**Pull request merged**